first_ruby/yarn.lock

```diff
@@ -0,0 +1,6888 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```

Comment:
Whats this file for?

Reply:
To manage dependency of specific version the project run on, this is generated after the dependency declared in `package.json`

Reply:
HOw is this different than `package.json`?

Reply:
`yarn.lock` records exact installed version in a lock file, which guarantee that if the project is run in another machine, it will install exact same package.